app/src/main/java/com/example/admin/lesson5service/MainActivity.java

```diff
@@ -1,13 +1,195 @@
+    public static List<Song> mSongs = new ArrayList<>();
+    public static final int MESSAGE_DELAY = 500;
+    public static final int NEXT_SONG = 1;
+    public static final int PREVIOUS_SONG = -1;
 
-public class MainActivity extends AppCompatActivity {
+    private Handler mHandler = new Handler() {
+        @Override
+        public void handleMessage(Message msg) {
+            super.handleMessage(msg);
+            int currentIndex = mService.getCurrentIndex();
+            int currentPosition = mService.getCurrentPosition();
+            mSeekBar.setProgress(currentPosition);
+            mTextTitle.setText(mSongs.get(currentIndex).getName());
+            mCurrentSongTime.setText(convertMilisecondToFormatTime(currentPosition));
+            mHandler.sendMessageDelayed(new Message(), MESSAGE_DELAY);
+        }
+    };
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        addData();
+        initViews();
+        Intent intent = new Intent(this, MusicService.class);
+        if (mService == null) {
+            startService(intent);
+        }
+        bindService(intent, mConnection, BIND_AUTO_CREATE);
+    }
+
+    private void addData() {
+        mSongs.clear();
+        mSongs.add(new Song("Chuyện anh vẫn chưa kể", "Chi Dân", R.raw.chuyen_anh_van_chua_ke));
+        mSongs.add(new Song("Chuyến đi của năm", "Soobin Hoàng Sơn", R.raw.chuyen_di_cua_nam));
+        mSongs.add(new Song("Đừng xin lỗi nữa", "Erik x Min", R.raw.dung_xin_loi_nua));
+        mSongs.add(new Song("Kém duyên", "Rum x Nit x Masew", R.raw.kem_duyen));
+        mSongs.add(new Song("Người lạ ơi", "Karik x Orange", R.raw.nguoi_la_oi));
+    }
+
+    private void initViews() {
+        mRecyclerMusic = findViewById(R.id.recycler_music);
+        mRecyclerMusic.setHasFixedSize(true);
+        SongAdapter musicAdapter = new SongAdapter(mSongs, this);
+        mRecyclerMusic.setAdapter(musicAdapter);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(this,
+                LinearLayoutManager.VERTICAL, false);
+        mRecyclerMusic.setLayoutManager(layoutManager);
+        mButtonNext = findViewById(R.id.image_button_next);
+        mButtonPrev = findViewById(R.id.image_button_prev);
+        mButtonPlay = findViewById(R.id.image_button_play);
+        mButtonPause = findViewById(R.id.image_button_pause);
+        mButtonPlay.setOnClickListener(this);
+        mButtonNext.setOnClickListener(this);
+        mButtonPrev.setOnClickListener(this);
+        mButtonPause.setOnClickListener(this);
+        mTextTitle = findViewById(R.id.text_title);
+        mCurrentSongTime = findViewById(R.id.text_time_start);
+        mDurationSongTime = findViewById(R.id.text_time_end);
+        mSeekBar = findViewById(R.id.seek_bar_song);
+
+        mSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
```

Comment:
Không sử dụng anonymous mà nên implement ra ngoài
```suggestion
        mSeekBar.setOnSeekBarChangeListener(this);
```